src/cmake_build_extension/cmake_extension.py

```diff
@@ -34,6 +35,7 @@ def __init__(
         cmake_build_type: str = "Release",
         cmake_component: str = None,
         cmake_depends_on: List[str] = (),
+        cmake_parallel_jobs: int = 1,
```

Comment:
```suggestion
        cmake_parallel_jobs: int = None,
```

In order to use the native parallelism of Ninja, maybe we can leave the default generators behavior as default. I'd pass `-jX` to CMake only if the user explicitly wants to pass this option.

Note that on MSVC, job parallelism is not as straightforward as on Linux/macOS. It could be done independently from `cmake_build_extension` by [properly setting the right compile options](https://discourse.cmake.org/t/parallel-does-not-really-enable-parallel-compiles-with-msbuild/964/10) in downstream CMake projects.